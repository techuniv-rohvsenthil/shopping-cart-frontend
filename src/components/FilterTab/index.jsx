import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';
import Button from '../Button';

const FilterTab = (props) => {
  const { data, setter } = props;
  if (!data.length) {
    return (
      <div />
    );
  }
  const tabs = data.map((tab) => (
    <Button buttonTestID="test-btn" setter={setter}>{tab}</Button>
  ));
  return (
    <div className={styles.tab}>
      {tabs}
    </div>
  );
};

FilterTab.propTypes = {
  data: propTypes.arrayOf(propTypes.string).isRequired,
};


export default FilterTab;
