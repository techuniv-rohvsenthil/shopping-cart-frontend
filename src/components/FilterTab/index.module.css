.tab{
    display: inline-block;
    background-color: #bdbdbd;
    margin: 2rem;
    padding: 0.3rem;
    border-radius: 1rem;
}