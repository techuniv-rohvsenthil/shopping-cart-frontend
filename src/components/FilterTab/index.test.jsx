import React from 'react';
import { render } from '@testing-library/react';
import FilterTab from './index';

const prodArray = [
    {
    prodId: '1',
    prodName: "Apple - 1kg",
    prodPrice: '210',
    prodQuantity: '10',
    prodImage: 'https://techunic-eval4.s3.amazonaws.com/apple.jpg',
    category: 'cat1',
  },
  
  {
    prodId: '2',
    prodName: "Apple - 1kg",
    prodPrice: '210',
    prodQuantity: '10',
    prodImage: 'https://techunic-eval4.s3.amazonaws.com/apple.jpg',
    category: 'cat1',
  },
  
  {
    prodId: '3',
    prodName: "Apple - 1kg",
    prodPrice: '210',
    prodQuantity: '10',
    prodImage: 'https://techunic-eval4.s3.amazonaws.com/apple.jpg',
    category: 'cat1',
  },
  {
    prodId: '4',
    prodName: "Apple - 1kg",
    prodPrice: '210',
    prodQuantity: '10',
    prodImage: 'https://techunic-eval4.s3.amazonaws.com/apple.jpg',
    category: 'cat1',
  },
  ];

describe('the FilterTab component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<FilterTab  data={prodArray}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});