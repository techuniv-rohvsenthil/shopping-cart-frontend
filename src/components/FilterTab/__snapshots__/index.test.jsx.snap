// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FilterTab component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="test-btn"
      type="button"
    >
      cat1
    </button>
    <button
      data-testid="test-btn"
      type="button"
    >
      cat2
    </button>
    <button
      data-testid="test-btn"
      type="button"
    >
      cat3
    </button>
  </div>
</DocumentFragment>
`;
